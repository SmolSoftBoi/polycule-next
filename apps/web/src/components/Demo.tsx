'use client';

import type { FC } from 'react';

/** Props for {@link Demo}. */
export interface DemoProps {
  /** Message to display. */
  message: string;
}

/**
 * Renders a simple demo block showing the message.
 */
export const Demo: FC<DemoProps> = ({ message }) => {
  return (
    <div className="rounded border border-dashed border-gray-400 p-4 text-center">
      {message} ðŸ˜Š
    </div>
  );
};
