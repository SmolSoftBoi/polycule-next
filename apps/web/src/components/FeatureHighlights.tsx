'use client';

import type { FC, ReactNode } from 'react';

/**
 * Feature item structure for {@link FeatureHighlights}.
 */
export interface Feature {
  /** Emoji or icon to represent the feature. */
  icon: ReactNode;
  /** Description of the feature. */
  text: string;
}

/** Props for {@link FeatureHighlights}. */
export interface FeatureHighlightsProps {
  /** Array of features to display. */
  features: Feature[];
}

/**
 * Lists the application's key features using icons.
 */
export const FeatureHighlights: FC<FeatureHighlightsProps> = ({ features }) => {
  return (
    <ul className="grid gap-4">
      {features.map((feature, index) => (
        <li className="flex items-start gap-2" key={index}>
          <span className="text-xl" aria-hidden>
            {feature.icon}
          </span>
          <span>{feature.text}</span>
        </li>
      ))}
    </ul>
  );
};
