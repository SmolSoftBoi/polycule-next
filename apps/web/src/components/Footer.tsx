'use client';

import Link from 'next/link';
import type { FC } from 'react';

/** A single footer link with label and URL. */
export interface FooterLink {
  /** Link destination. */
  href: string;
  /** Link text. */
  label: string;
}

/** Props for {@link Footer}. */
export interface FooterProps {
  /** Navigation links displayed in the footer. */
  links: FooterLink[];
}

/**
 * Renders the application footer with copyright and links.
 */
export const Footer: FC<FooterProps> = ({ links }) => {
  const year = new Date().getFullYear();
  return (
    <footer className="flex flex-col items-center gap-2 text-sm text-gray-500">
      <nav className="flex flex-wrap justify-center gap-4">
        {links.map(link => (
          <Link
            key={link.href}
            className="hover:underline hover:underline-offset-4"
            href={link.href}
          >
            {link.label}
          </Link>
        ))}
      </nav>
      <span>Â© {year} Polymap</span>
    </footer>
  );
};
