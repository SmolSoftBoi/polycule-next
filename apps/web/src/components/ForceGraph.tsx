'use client';

import dynamic from 'next/dynamic';
import type { NodeObject, LinkObject } from 'force-graph';
import type { FC } from 'react';

const ForceGraph2D = dynamic(
  () => import('react-force-graph').then(mod => mod.ForceGraph2D),
  { ssr: false }
);

/** Props for {@link ForceGraph}. */
export interface ForceGraphProps {
  data: { nodes: NodeObject[]; links: LinkObject[] };
}

/**
 * Renders an interactive force-directed graph.
 */
export const ForceGraph: FC<ForceGraphProps> = ({ data }) => {
  return <ForceGraph2D graphData={data} />;
};
