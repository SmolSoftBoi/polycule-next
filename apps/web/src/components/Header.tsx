'use client';

import Link from 'next/link';
import type { FC } from 'react';

/** Props for {@link Header}. */
export interface HeaderProps {
  /** Title shown in the header. */
  title: string;
}

/**
 * Renders the application header with a home link.
 */
export const Header: FC<HeaderProps> = ({ title }) => {
  return (
    <header className="bg-background px-4 py-2 text-foreground">
      <Link className="text-xl font-bold no-underline" href="/">
        {title}
      </Link>
    </header>
  );
};

