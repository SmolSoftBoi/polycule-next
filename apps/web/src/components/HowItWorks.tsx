'use client';

import type { FC, ReactNode } from 'react';

/**
 * Step information for {@link HowItWorks}.
 */
export interface Step {
  /** Icon representing the step. */
  icon: ReactNode;
  /** Title of the step. */
  title: string;
  /** Detailed explanation. */
  description: string;
}

/** Props for {@link HowItWorks}. */
export interface HowItWorksProps {
  /** Steps describing the process. */
  steps: Step[];
}

/**
 * Explains how the application works using a simple list of steps.
 */
export const HowItWorks: FC<HowItWorksProps> = ({ steps }) => {
  return (
    <ol className="grid list-inside list-decimal gap-6">
      {steps.map((step, index) => (
        <li className="flex items-start gap-2" key={index}>
          <span className="text-xl" aria-hidden>
            {step.icon}
          </span>
          <div>
            <div className="font-semibold">{step.title}</div>
            <p className="text-sm">{step.description}</p>
          </div>
        </li>
      ))}
    </ol>
  );
};
