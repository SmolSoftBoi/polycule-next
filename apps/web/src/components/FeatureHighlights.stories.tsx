import type { Meta, StoryObj } from '@storybook/react';

import { FeatureHighlights } from './FeatureHighlights';

const meta: Meta<typeof FeatureHighlights> = {
  component: FeatureHighlights,
  title: 'Components/FeatureHighlights',
};
export default meta;

type Story = StoryObj<typeof FeatureHighlights>;

export const Default: Story = {
  args: {
    features: [
      { icon: 'ðŸš€', text: 'Fast' },
      { icon: 'ðŸ”’', text: 'Secure' },
      { icon: 'âš¡', text: 'Efficient' },
    ],
  },
};
