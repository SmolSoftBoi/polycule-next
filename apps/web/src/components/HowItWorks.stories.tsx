import type { Meta, StoryObj } from '@storybook/react';

import { HowItWorks } from './HowItWorks';

const meta: Meta<typeof HowItWorks> = {
  component: HowItWorks,
  title: 'Components/HowItWorks',
};
export default meta;

type Story = StoryObj<typeof HowItWorks>;

export const Default: Story = {
  args: {
    steps: [
      {
        icon: 'üíæ',
        title: 'Upload data',
        description: 'Import your dataset in seconds.',
      },
      {
        icon: 'üó∫Ô∏è',
        title: 'Visualize',
        description: 'Explore relationships on an interactive map.',
      },
      {
        icon: 'ü§ù',
        title: 'Collaborate',
        description: 'Share insights with your team.',
      },
    ],
  },
};
