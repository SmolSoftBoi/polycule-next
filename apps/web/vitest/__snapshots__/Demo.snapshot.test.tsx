import { render } from '@testing-library/react';

import { Demo } from '../../src/components/Demo';

describe('Demo snapshot', () => {
  it('matches', () => {
    const { container } = render(<Demo message="Snapshot" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
