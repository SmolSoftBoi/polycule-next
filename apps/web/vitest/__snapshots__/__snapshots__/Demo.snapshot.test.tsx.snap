// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Demo snapshot > matches 1`] = `
<div
  class="rounded border border-dashed border-gray-400 p-4 text-center"
>
  Snapshot
   ğŸ˜Š
</div>
`;
